### ISSUE TITLE
Enabling the SLiM display manager is stateful and breaks later enabling the xpra DM

### ISSUE THREAD

**Issue by thblt** [NixOS/nixpkgs#44941]
## Issue description

TL;DR Enabling the SLiM display manager, then replacing it by xpra, fails on `nixos-rebuild switch` with an error message suggesting that SLiM is still enabled.  Explicitely disabling SLiM then rebuilding solves the issue, the explicit disable can be removed after a successful rebuild, which suggests some state persists between rebuilds.

### Steps to reproduce

1. Add the following line to `/etc/nix/configuration.nix`:

   ```
   services.displayManager.slim.enable = true;
   ```

2 . Run `nixos-rebuild switch`.

3. Modify this line to:

   ```
   services.displayManager.xpra.enable = true;
   ```

4. Run `nixos-rebuild switch`, again.  It will fail with the following log:

    ```
    error: The unique option `services.xserver.displayManager.job.execCmd' 
    is defined multiple times, in 
    `/nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/services/x11/display-managers/xpra.nix'
    and 
    `/nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/services/x11/display-managers/slim.nix'.                                       
    ```

 5. Add the following line to explicitely disable Slim:

   ```
   services.displayManager.slim.enable = false;
   ```

6. Run `nixos-rebuild switch`, again.  It works now.

7. Remove the line you added at step 5.

8. Run `nixos-rebuild switch`, again.  It still works.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.17.14, NixOS, 18.09pre148970.b5b11e9da43 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.4`
 - channels(thblt): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by dezgeg** [comment]
There's nothing stateful, it's just that the default value of `services.displayManager.slim.enable` is `true`.

IIRC there's similar other things in the X server module, for instance `xterm` the desktop manager is enabled by default. Would be nice to get these fixed.

**Comment by thblt** [comment]
> it's just that the default value of services.displayManager.slim.enable is true.

According to the options search engine, that's not the case, but it may not be up-to-date.  But there's still a bit of state, because you only need to explicitely disable SLiM *once*.  Once nixos-rebuild succeeds, the explicit disable can be removed from configuration.nix.

To be clear, this sequence

**Comment by prusnak** [comment]
SLIM is no longer in nixpkgs: ce26b3e
